### PR TITLE
Amit/collection modal

### PR DESCRIPTION
**Pull request by ak130903**
image

### CODE REVIEW

File: src/components/base/CollectionModal/CollectionModal.style.tsx [hunk 1]
```diff
@@ -0,0 +1,98 @@
+import { Modal, styled } from '@mui/material';
+import { DocumetsIcon } from '../../icons';
+import { PersonalCollectionIcon } from '../../icons/PersonalCollectionIcon';
+
+
+
```
Comment (ShakedZrihen) [inline src/components/base/CollectionModal/CollectionModal.style.tsx:6]: remove, 1 new line is enough

File: src/components/base/CollectionModal/CollectionModal.style.tsx [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+export const StyledColletionModal = styled(Modal)`
+  display: flex;
+  justify-content: center;
+  align-items: center; 
+  overflow: auto; 
+  width: 100%;
+  border-radius: 12px;
```
Comment (ShakedZrihen) [inline src/components/base/CollectionModal/CollectionModal.style.tsx:13]: use rem and not div.. here is 0.8rem +-

File: src/components/base/CollectionModal/CollectionModal.style.tsx [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+  flex-direction: row;
+  align-items: center;
+  height: 48px;
+  border: none; 
+  border-radius: 8px;
+  cursor: pointer;
+  transition: background-color 0.3s ease, border-color 0.01s ease;
+  padding: 10px; 
+  box-shadow: 0px 4px 4px 0px rgba(0, 0, 0, 0.25);
+  background-color: ${({ isSelected }) => (isSelected ? '#e3f2fd' : '#fff')};
```
Comment (ShakedZrihen) [inline src/components/base/CollectionModal/CollectionModal.style.tsx:84]: please use colorPalette and not hard coded colors

File: src/components/base/CollectionModal/CollectionModal.style.tsx [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+  border: none; 
+  border-radius: 8px;
+  cursor: pointer;
+  transition: background-color 0.3s ease, border-color 0.01s ease;
+  padding: 10px; 
+  box-shadow: 0px 4px 4px 0px rgba(0, 0, 0, 0.25);
+  background-color: ${({ isSelected }) => (isSelected ? '#e3f2fd' : '#fff')}; 
+
+  &:active {
+    background-color: #d0d0d0;
```
Comment (ShakedZrihen) [inline src/components/base/CollectionModal/CollectionModal.style.tsx:87]: please use colorPalette and not hard coded colors

File: src/components/base/CollectionModal/CollectionModal.tsx [hunk 5]
```diff
@@ -0,0 +1,247 @@
...
+  postId?: string;
+  collections: {
+    id: number | string;
+    name: string;
+    private: boolean;
+    lastUpdate: Date;
+  }[];
+}
+
+const handleCollectionClick = (
```
Comment (ShakedZrihen) [inline src/components/base/CollectionModal/CollectionModal.tsx:39]: please put it under the CollectionModal as it used state functions

File: src/components/base/CollectionModal/CollectionModal.tsx [hunk 6]
```diff
@@ -0,0 +1,247 @@
...
+  collections,
+  isOpen,
+  onClose,
+  postId,
+}: ColletionModalProps) {
+  const [searchTerm, setSearchTerm] = useState('');
+  const [selectedCollections, setSelectedCollections] = useState<
+    (number | string)[]
+  >([]);
+  const [activeTab, setActiveTab] = useState(3);
```
Comment (ShakedZrihen) [inline src/components/base/CollectionModal/CollectionModal.tsx:63]: please use enum for the tabs,

for example

```
enum Tabs {
 AllCollection = 1,
UserCollections = 2,
SharedCollections = 3
}
```

File: src/components/base/CollectionModal/CollectionModal.tsx [hunk 7]
```diff
@@ -0,0 +1,247 @@
...
+  ).length;
+  const publicCollectionsCount = collections.filter(
+    collection => !collection.private,
+  ).length;
+
+  const filteredCollections = collections.filter(collection => {
+    const matchesSearchTerm = collection.name
+      .toLowerCase()
+      .includes(searchTerm.toLowerCase());
+    if (activeTab === 2) return matchesSearchTerm && collection.private;
```
Comment (ShakedZrihen) [inline src/components/base/CollectionModal/CollectionModal.tsx:81]: use your new enum here

Comment (ShakedZrihen) [inline src/components/base/CollectionModal/CollectionModal.tsx:81]:
```suggestion
    if (activeTab === 2) {
        return matchesSearchTerm && collection.private;
    }
```

File: src/components/base/CollectionModal/CollectionModal.tsx [hunk 8]
```diff
@@ -0,0 +1,247 @@
...
+    if (activeTab === 1) return matchesSearchTerm && !collection.private;
```
Comment (ShakedZrihen) [inline src/components/base/CollectionModal/CollectionModal.tsx:82]:
```suggestion
    if (activeTab === 1) {
         return matchesSearchTerm && !collection.private;
    }
```

File: src/components/base/CollectionModal/CollectionModal.tsx [hunk 9]
```diff
@@ -0,0 +1,247 @@
...
+    return matchesSearchTerm;
+  });
+
+  const handleConfirm = () => {
+    console.log('Selected Collections IDs:', selectedCollections);
+    setSelectedCollections([]);
+    if (onClose) onClose();
```
Comment (ShakedZrihen) [inline src/components/base/CollectionModal/CollectionModal.tsx:89]: please add parentheses to all "ifs"

File: src/components/base/CollectionModal/CollectionModal.tsx [hunk 10]
```diff
@@ -0,0 +1,247 @@
...
+    const matchesSearchTerm = collection.name
+      .toLowerCase()
+      .includes(searchTerm.toLowerCase());
+    if (activeTab === 2) return matchesSearchTerm && collection.private;
+    if (activeTab === 1) return matchesSearchTerm && !collection.private;
+    return matchesSearchTerm;
+  });
+
+  const handleConfirm = () => {
+    console.log('Selected Collections IDs:', selectedCollections);
```
Comment (ShakedZrihen) [inline src/components/base/CollectionModal/CollectionModal.tsx:87]: remove log

File: src/components/base/CollectionModal/CollectionModal.tsx [hunk 11]
```diff
@@ -0,0 +1,247 @@
...
+            style={{
+              fontSize: '21px',
+              fontWeight: 700,
+              lineHeight: '28px',
+              textAlign: 'right',
+              padding: '6px',
+            }}
+          >
+            שמירה לאוסף
+          </Typography>
```
Comment (ShakedZrihen) [inline src/components/base/CollectionModal/CollectionModal.tsx:119]: use StyledTypography and not inline style

extract text to translation, you can ask yuval or maxim or search for "t('" and see how they did it

File: src/components/base/CollectionModal/CollectionModal.tsx [hunk 12]
```diff
@@ -0,0 +1,247 @@
...
+              }}
+              InputProps={{
+                endAdornment: (
+                  <InputAdornment position='end'>
+                    <SearchIcon />
+                  </InputAdornment>
+                ),
+                sx: {
+                  borderRadius: '20px',
+                  height: '40px',
```
Comment (ShakedZrihen) [inline src/components/base/CollectionModal/CollectionModal.tsx:143]: use rem and not px, 1 rem is around 13px

File: src/components/base/CollectionModal/CollectionModal.tsx
Comment (ShakedZrihen) [inline src/components/base/CollectionModal/CollectionModal.tsx:1]: this file is a bit long, consider split it to smaller component

File: src/components/custom/Post/Post.tsx [hunk 13]
```diff
@@ -62,11 +63,25 @@ export function Post(props: PostProps & { className?: string }) {
...
+  };
+
+  
+  const [content, setContent] = useState<string | undefined>(selected || original);
+  // Close modal handler
+  const handleCloseModal = () => {
+    setIsCollectionModalOpen(false);
+  };
+
+
```
Comment (ShakedZrihen) [inline src/components/custom/Post/Post.tsx:83]: I think it redundant, you can delete it

File: src/components/base/CollectionModal/CollectionModal.style.tsx [hunk 14]
```diff
@@ -0,0 +1,116 @@
...
+export const ActionsContainer = styled('div')`
+  display: flex;
+  justify-content: flex-start; 
+  align-items: center; 
+  gap: 10px;
+  position: absolute; 
+  bottom: 1rem; 
+  left: 1rem;
+  padding-top: 1rem;
+  border-top: 1px solid #e0e0e0;
```
Comment (ShakedZrihen) [inline src/components/base/CollectionModal/CollectionModal.style.tsx:47]: please use colorPalette

File: src/components/icons/PersonalCollectionIcon/PersonalCollectionIcon.tsx [hunk 15]
```diff
@@ -0,0 +1,58 @@
...
+import { StyledIcon } from '../StyledIcon';
+
+export function PersonalCollectionIcon(props: IconProps) {
+  const { size = 30, color = '#2860a8', className } = props;
+
+  return (
+    <StyledIcon
+      data-testid='specter-personal-collection-icon-icon'
+      className={className}
+      style={{ width: size, height: size, color: 'white' }}
```
Comment (ShakedZrihen) [inline src/components/icons/PersonalCollectionIcon/PersonalCollectionIcon.tsx:11]:
```suggestion
      style={{ width: size, height: size, color }}
```

**Review by supermario1236565 — DISMISSED** [review]

**Pull request merged**